src/Azure.Functions.Cli/Diagnostics/ColoredConsoleLogger.cs

```diff
@@ -14,18 +14,53 @@ public class ColoredConsoleLogger : ILogger
         private readonly bool _verboseErrors;
         private readonly string _category;
         private readonly LoggingFilterHelper _loggingFilterHelper;
+        private readonly LoggerFilterOptions _loggerFilterOptions;
         private readonly string[] allowedLogsPrefixes = new string[] { "Worker process started and initialized.", "Host lock lease acquired by instance ID" };
+        private static readonly LoggerRuleSelector RuleSelector = new LoggerRuleSelector();
+        private static readonly Type ProviderType = typeof(ColoredConsoleLoggerProvider);
 
-        public ColoredConsoleLogger(string category, LoggingFilterHelper loggingFilterHelper)
+        public ColoredConsoleLogger(string category, LoggingFilterHelper loggingFilterHelper, LoggerFilterOptions loggerFilterOptions)
         {
             _category = category;
-            _loggingFilterHelper = loggingFilterHelper;
+            _loggerFilterOptions = loggerFilterOptions ?? throw new ArgumentNullException(nameof(loggerFilterOptions));
+            _loggingFilterHelper = loggingFilterHelper ?? throw new ArgumentNullException(nameof(loggingFilterHelper));
             _verboseErrors = StaticSettings.IsDebug;
         }
 
+        internal LoggerFilterRule SelectRule(string categoryName, LoggerFilterOptions loggerFilterOptions)
+        {
+            RuleSelector.Select(loggerFilterOptions, ProviderType, categoryName,
+                out LogLevel? minLevel, out Func<string, string, LogLevel, bool> filter);
+
+            return new LoggerFilterRule(ProviderType.FullName, categoryName, minLevel, filter);
+        }
+
+        internal bool IsEnabled(string category, LogLevel logLevel)
+        {
+            LoggerFilterRule filterRule = SelectRule(category, _loggerFilterOptions);
+
+            if (filterRule.LogLevel != null && logLevel < filterRule.LogLevel)
+            {
+                return false;
+            }
+            if (filterRule.Filter != null)
+            {
+                bool enabled = filterRule.Filter(ProviderType.FullName, category, logLevel);
+                if (!enabled)
+                {
+                    return false;
+                }
+            }
```

Comment:
For these two steps, if the filter comes back as true, we keep going? Shouldn't we return back if there's an explicit filter or log level set?

Reply:
- If filter returns `false`  - it returns  - that means it failed check for `DefaultWebJobsFilter` added at webhost level.
- Filter comes back as `true` - for system logs from jobhost level - if loglevel is overwritten in host.json / env vars - `LogLevel` on filter is not null - this log level is used instead of defaults.

Do let me know if this is not clear.